### ISSUE TITLE
Type lost in serialization

### ISSUE THREAD

**Issue by stoffen** [r3labs/diff#73]
Hi, this might not be a bug, but an issue I´ve ran into and need to work around. Consider the following 

```
type Test struct {
	S *int `json:"s,omitempty"`
}

func TestPointerDiff(t *testing.T) {
	val1 := 1
	val2 := 2

	t1 := Test{S: &val1}
	t2 := Test{S: &val2}

	changelog, err := diff.Diff(t1, t2)
	assert.NoError(t, err)

	js, err := json.Marshal(changelog)
	assert.NoError(t, err)

	assert.NoError(t, json.Unmarshal(js, &changelog))

	patchLog := diff.Patch(changelog, &t1)
	assert.False(t, patchLog.HasErrors())
}
```

This will fail, since the type of the int is lost when marshaling to JSON. 

Any ideas how I could work around this?

**Comment by stoffen** [comment]
For now I just persist the type of the `To`, and handle that after unmarshaling from JSON

```
type Change struct {
	diff.Change
	ToType string `json:"toType"`
}
```

**Comment by purehyperbole** [comment]
Hey @stoffen ,

Thanks for raising the issue! We actually have a `ConvertCompatibleTypes()` option for this very problem.

I've just checked and it wasn't working with pointer values, but I've just applied the same fix from `v2.14.3` to that path, so it should be working now if you update to `v2.14.4`. 

You can patch with that option like so:
```go
d, err := diff.NewDiffer(diff.ConvertCompatibleTypes())
...

patchLog := d.Patch(changelog, &t1)
```

Please let me know if that fixes your issue

**Comment by stoffen** [comment]
Yes I did try the option for the differ but as you found it did not handle the pointers. Thanks:)
